### PR TITLE
add multi-gateways support to api command and plan support to declarative command

### PR DESCRIPTION
**Pull request by jquantin**
Hi,

This is 2 things : 

- a small modification of the api command to use more than one gateway to create an API.
    I only add a new param "gateways" (with a s) and let the old one "gateway" (without s) to be sure to not break things. I can merge the 2 params together (the one without s) if its preferable.
- the support for plans on the declaratives command. Works with a "plans" section on the root of organisation and with the sames policies declarations than those inside api.

thanks

### CODE REVIEW

File: src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java [hunk 1]
```diff
@@ -62,6 +67,9 @@
     @Option(name = "--gateway", aliases = {"-g"}, usage = "Gateway")
```
Comment (outofcoffee) [inline src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java:67]: Can we make the `gateway` argument a `List<String>`? Args4j will do the rest.

File: src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java [hunk 2]
```diff
@@ -81,11 +89,23 @@ public void performAction(CmdLineParser parser) throws CommandException {
                 description,
                 initialVersion);
 
+        final List gatewaysList = Lists.newArrayList();
```
Comment (outofcoffee) [inline src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java]: This won't be needed anymore if we make the private field a `List<String>`

File: src/main/java/io/apiman/cli/command/plan/model/PlanVersion.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class PlanVersion {
+    @JsonProperty
+    private String version;
+
+    /**
+     * Never clone a previous version when creating a new version.
+     */
+    @JsonProperty
+    final private boolean clone = false;
```
Comment (outofcoffee) [inline src/main/java/io/apiman/cli/command/plan/model/PlanVersion.java:38]: Is this needed for plan creation?

File: src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java (hunk 1 above)
Comment (jquantin) [reply src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java:67]: ok I remove the "gateways" arg and make the "gateway" one, being a List (can occurs more than once)
for instance : "-g gw1 -g gw2"

File: src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java (hunk 2 above)
Comment (jquantin) [reply src/main/java/io/apiman/cli/command/api/command/ApiCreateCommand.java]: I steel use it for creating a list of GatewayApi Objects

File: src/main/java/io/apiman/cli/command/plan/model/PlanVersion.java (hunk 3 above)
Comment (jquantin) [reply src/main/java/io/apiman/cli/command/plan/model/PlanVersion.java:38]: Yes it works like the POST of api's version (@POST("/organizations/{orgName}/plans/{planName}/versions")
The clone's mecanism is the same as Api one.

**Comment by msavy** [comment]
Hi @jquantin I haven't forgotten about this. As soon as my main PR is resolved, I'll be looking to get your work in (I'll refactor it).

**Comment by jquantin** [comment]
Cool !
I started to merge with your attempt on this branch : https://github.com/jquantin/apiman-cli/tree/OrgAPI
And I add a lot more inside : work to serialize an organization with clients, contracts, plans and apis (use the manager api) and other works to use this on entry of declarative Command. I think that it could be compatible with your description here : https://github.com/apiman/apiman-cli/pull/28#issuecomment-341922984.
Don't hesitate to call or mail me.

Thank-you

**Comment by msavy** [comment]
Hey @jquantin is the branch you reference (OrgAPI) still the latest? Looking to merge in your work on my branch.

**Comment by jquantin** [comment]
Hi,
Yes this is my last tag. All i have done is in this one.
Thanks

**Comment by outofcoffee** [comment]
Superseded by #36

**Pull request closed without merging**